packages/wordcount/src/index.js

```diff
@@ -17,18 +17,26 @@ import stripShortcodes from './stripShortcodes';
 import stripSpaces from './stripSpaces';
 import transposeHTMLEntitiesToCountableChars from './transposeHTMLEntitiesToCountableChars';
 
+/** @typedef {import('./defaultSettings').WPWordCountSettings}  WPWordCountSettings */
+
+/**
+ * Possible ways of counting.
+ *
+ * @typedef {'words'|'characters_excluding_spaces'|'characters_including_spaces'} WPWordCountStrategy
+ */
+
 /**
  * Private function to manage the settings.
  *
- * @param {string} type         The type of count to be done.
- * @param {Object} userSettings Custom settings for the count.
+ * @param {WPWordCountStrategy} type         The type of count to be done.
+ * @param {WPWordCountSettings} userSettings Custom settings for the count.
  *
- * @return {void|Object|*} The combined settings object to be used.
+ * @return {WPWordCountSettings} The combined settings object to be used.
```

Comment:
It seems appealing to have a narrower return type here. The `Partial<…>` type is good for the inputs, all of the fields are optional. However, we have better guarantees on the return type. Something like `{WPWordCountSettings & typeof defaultSettings}` might be a good start, that _should_ be the type of `const settings = extend( defaultSettings, userSettings )` a few lines below.

Reply:
Thanks, @sirreal – see ffb27cc and 4576c98 for an exploration in this area.